### ISSUE TITLE
repository-hdfs tests fail in a fips java runtime

### ISSUE THREAD

**Issue by jkakavas** [elastic/elasticsearch#40079]
Example: 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.0+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/73/console

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.7+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/73/console

```
org.apache.hadoop.ipc.RemoteException(org.apache.hadoop.security.AccessControlException): Unable to find SASL server implementation for DIGEST-MD5
02:52:12     	at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1481)
02:52:12     	at org.apache.hadoop.ipc.Client.call(Client.java:1427)
02:52:12     	at org.apache.hadoop.ipc.Client.call(Client.java:1337)
02:52:12     	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:227)
02:52:12     	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:116)
02:52:12     	at com.sun.proxy.$Proxy18.getDatanodeReport(Unknown Source)
02:52:12     	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getDatanodeReport(ClientNamenodeProtocolTranslatorPB.java:641)
02:52:12     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
02:52:12     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
02:52:12     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
02:52:12     	at java.lang.reflect.Method.invoke(Method.java:498)
02:52:12     	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:398)
02:52:12     	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:163)
02:52:12     	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:155)
02:52:12     	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
02:52:12     	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:335)
02:52:12     	at com.sun.proxy.$Proxy19.getDatanodeReport(Unknown Source)
02:52:12     	at org.apache.hadoop.hdfs.DFSClient.datanodeReport(DFSClient.java:2161)
02:52:12     	at org.apache.hadoop.hdfs.MiniDFSCluster.waitActive(MiniDFSCluster.java:2459)
02:52:12     	at org.apache.hadoop.hdfs.MiniDFSCluster.waitActive(MiniDFSCluster.java:2503)
02:52:12     	at org.apache.hadoop.hdfs.MiniDFSCluster.startDataNodes(MiniDFSCluster.java:1537)
02:52:12     	at org.apache.hadoop.hdfs.MiniDFSCluster.initMiniDFSCluster(MiniDFSCluster.java:847)
02:52:12     	at org.apache.hadoop.hdfs.MiniDFSCluster.<init>(MiniDFSCluster.java:483)
02:52:12     	at org.apache.hadoop.hdfs.MiniDFSCluster$Builder.build(MiniDFSCluster.java:442)
02:52:12     	at hdfs.MiniHDFS.main(MiniHDFS.java:120)
```

This happens because the `hdfs-fixture` is run with `runtime.java` which might be a JVM in FIPS approved only mode, where `DIGEST-MD5` is unavailable

**Comment by benwtrent** [comment]
Found a related hdfs repository error. Unsure if it is caused by the fix for this issue or not:

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.3+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/28/console

yaml hdfs repository test fail for the following reason:

```
[2019-07-17T12:03:42,641][INFO ][o.e.r.RepositoriesService] [integTest-0] put repository [test_snapshot_repository]
[2019-07-17T12:03:42,708][WARN ][o.a.h.u.NativeCodeLoader ] [integTest-0] Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2019-07-17T12:03:43,215][WARN ][o.a.h.i.Client           ] [integTest-0] Failed to connect to server: localhost/127.0.0.1:9999: try once and fail.
java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717) ~[?:?]
	at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:206) ~[?:?]
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:531) ~[?:?]
	at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:495) ~[?:?]
	at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:681) ~[?:?]
	at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:777) ~[?:?]
	at org.apache.hadoop.ipc.Client$Connection.access$3500(Client.java:409) ~[?:?]
	at org.apache.hadoop.ipc.Client.getConnection(Client.java:1542) ~[?:?]
	at org.apache.hadoop.ipc.Client.call(Client.java:1373) ~[?:?]
	at org.apache.hadoop.ipc.Client.call(Client.java:1337) ~[?:?]
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:227) ~[?:?]
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:116) ~[?:?]
	at com.sun.proxy.$Proxy36.mkdirs(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.mkdirs(ClientNamenodeProtocolTranslatorPB.java:574) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_212]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:398) ~[?:?]
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:163) ~[?:?]
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:155) ~[?:?]
	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95) ~[?:?]
	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:335) ~[?:?]
	at com.sun.proxy.$Proxy37.mkdirs(Unknown Source) ~[?:?]
	at org.apache.hadoop.hdfs.DFSClient.primitiveMkdir(DFSClient.java:2525) ~[?:?]
	at org.apache.hadoop.fs.Hdfs.mkdir(Hdfs.java:311) ~[?:?]
	at org.apache.hadoop.fs.FileContext$4.next(FileContext.java:738) ~[?:?]
	at org.apache.hadoop.fs.FileContext$4.next(FileContext.java:734) ~[?:?]
	at org.apache.hadoop.fs.FSLinkResolver.resolve(FSLinkResolver.java:90) ~[?:?]
	at org.apache.hadoop.fs.FileContext.mkdir(FileContext.java:741) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsBlobStore.lambda$mkdirs$1(HdfsBlobStore.java:64) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsBlobStore.lambda$execute$2(HdfsBlobStore.java:114) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_212]
	at java.security.AccessController.doPrivileged(AccessController.java:713) ~[?:1.8.0_212]
	at org.elasticsearch.repositories.hdfs.HdfsSecurityContext.doPrivilegedOrThrow(HdfsSecurityContext.java:142) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsBlobStore.execute(HdfsBlobStore.java:112) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsBlobStore.mkdirs(HdfsBlobStore.java:63) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsBlobStore.<init>(HdfsBlobStore.java:55) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsRepository.createBlobstore(HdfsRepository.java:144) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsRepository.lambda$createBlobStore$1(HdfsRepository.java:232) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_212]
	at org.elasticsearch.repositories.hdfs.HdfsRepository.createBlobStore(HdfsRepository.java:231) ~[?:?]
	at org.elasticsearch.repositories.hdfs.HdfsRepository.createBlobStore(HdfsRepository.java:54) ~[?:?]
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.blobStore(BlobStoreRepository.java:338) ~[elasticsearch-7.3.0-SNAPSHOT.jar:7.3.0-SNAPSHOT]
	at org.elasticsearch.repositories.blobstore.BlobStoreRepository.startVerification(BlobStoreRepository.java:621) ~[elasticsearch-7.3.0-SNAPSHOT.jar:7.3.0-SNAPSHOT]
	at org.elasticsearch.repositories.RepositoriesService.lambda$verifyRepository$7(RepositoriesService.java:239) ~[elasticsearch-7.3.0-SNAPSHOT.jar:7.3.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:688) [elasticsearch-7.3.0-SNAPSHOT.jar:7.3.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_212]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by benwtrent** [comment]
Another related failure with Fips + repository-hdfs, ever since the merge of the fix that disables the tests in fips, on 7.2 build have continually failed with the same reason:

```
:plugins:repository-hdfs:testingConventions FAILED
--
Test classes are not included in any enabled task (:plugins:repository-hdfs:test,:plugins:repository-hdfs:integTestRunner):
* org.elasticsearch.repositories.hdfs.HaHdfsFailoverTestSuiteIT
```

All of the failures are on the 7.2 branch in the matrix-java-periodic job with the following parameters 

ES_BUILD_JAVA=openjdk12,
ES_RUNTIME_JAVA=java8fips,
nodes=general-purpose

All the failures occur on some flavor of linux.

Examples: 
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/78/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/85/console
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.2+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/86/console

**Comment by jkakavas** [comment]
resolved by #44283
